runner.py

```diff
@@ -1,12 +1,34 @@
-"""Script that is able to start Uvicorn-based REST API service."""
+"""Main app entrypoint. Starts Uvicorn-based REST API service."""
 
 import os
 
 import uvicorn
 
+
+def configure_hugging_face_envs() -> None:
+    """Configure HuggingFace library environment variables."""
+    # NOTE: We import config here to avoid triggering import of anything
+    # else via our code before other envs are set (mainly the gradio).
+    from ols.utils import config
+
+    cfg_file = os.environ.get("OLS_CONFIG_FILE", "olsconfig.yaml")
+    config.init_config(cfg_file)
+    if (
+        config.ols_config
+        and hasattr(config.ols_config, "reference_content")
+        and hasattr(config.ols_config.reference_content, "embeddings_model_path")
+        and config.ols_config.reference_content.embeddings_model_path
+    ):
+        os.environ["TRANSFORMERS_CACHE"] = (
+            config.ols_config.reference_content.embeddings_model_path
+        )
+        os.environ["TRANSFORMERS_OFFLINE"] = "1"
+
+
 if __name__ == "__main__":
-    # TODO syedriko consolidate this env var into a central location as per OLS-345.
     # disable Gradio analytics, which calls home to https://api.gradio.app
     os.environ["GRADIO_ANALYTICS_ENABLED"] = "false"
 
+    configure_hugging_face_envs()
```

Comment:
i would expect config initialization to be called from here, not from within configure_hugging_face_envs.  config initialization shouldn't be a side effect of setting the hugging face env vars.

Reply:
@onmete ^^^^